libflux/src/flux/semantic/flatbuffers/mod.rs

```diff
@@ -1,3 +1,1147 @@
 #![allow(clippy::all)]
 pub mod semantic_generated;
-pub mod types;
+mod types;
+
+use std::cell::RefCell;
+use std::rc::Rc;
+
+use crate::ast;
+
+use crate::semantic;
+use crate::semantic::walk;
+use flatbuffers::{UnionWIPOffset, WIPOffset};
+use semantic_generated::fbsemantic;
+
+extern crate chrono;
+use chrono::Offset;
+
+pub fn serialize(semantic_pkg: &mut semantic::nodes::Package) -> Result<(Vec<u8>, usize), String> {
+    let mut v = new_serializing_visitor_with_capacity(1024);
+    walk::walk(&mut v, Rc::new(walk::Node::Package(semantic_pkg)));
+    v.finish()
+}
+
+fn new_serializing_visitor_with_capacity<'a>(_capacity: usize) -> SerializingVisitor<'a> {
+    SerializingVisitor {
+        inner: Rc::new(RefCell::new(SerializingVisitorState::new_with_capacity(
+            1024,
+        ))),
+    }
+}
+
+struct SerializingVisitor<'a> {
+    inner: Rc<RefCell<SerializingVisitorState<'a>>>,
+}
+
+impl<'a> semantic::walk::Visitor<'_> for SerializingVisitor<'a> {
+    fn visit(&mut self, _node: Rc<walk::Node<'_>>) -> bool {
+        let v = self.inner.borrow();
+        if let Some(_) = &v.err {
+            return false;
+        }
+        Some(SerializingVisitor {
+            inner: Rc::clone(&self.inner),
+        });
```

Comment:
I wonder if you could explain this block to me, as I haven't done much with `Rc` itself just yet. It looks like we're creating a `SerializingVisitor` with an `inner` that is a strong reference counte of `self.inner`, right, but we're not holding a handle to the visitor, and adding it to a `Option` of `Some` but also not retaining a handle to it. It looks like this block would be a no-op short of incrementing the reference counter, so I think I'm missing something.

Reply:
Yeah, so the process followed by the SerializingVisitor is something that I borrowed from @wolffcm when he implemented this on the AST side. By `holding a handle` to the visitor, are you referring to the place where its dereferenced?

Reply:
_Should_ this return an `Option<Self>`? Is there a reason why the interface should/shouldn't be homogeneous?

Reply:
Hmm...I'm curious about that as well. I think I'm going to go ahead and merge this as is, but I can easily update it in my next PR. Is the idea here that we just need to remove the option?